sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2157,10 +2157,12 @@ cl_int ExecCGCommand::enqueueImp() {
       } else {
         assert(MQueue->getPlugin().getBackend() ==
                backend::ext_intel_esimd_emulator);
+        // Dims==0 for 'single_task() - void(void) type'
+        uint32_t Dims = (Args.size() > 0) ? NDRDesc.Dims : 0;
```

Comment:
AFAICT, single_task is not supported. Should this cause DIE_NO_IMPLEMENTATION?

Reply:
'single_task()' is widely used in intel/llvm-test-suite. `ESIMD/SYCL/printf.cpp` uses 'single_task()'. I would like to make sure if it is going to be supported or not.

Reply:
But your code implies it is not supported yet. So DIE_NO_IMPLEMENTATION should happen on attempt to use it.

Reply:
`DIE_NO_IMPLEMENTATION` is defined in `sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp`. Maybe execution attempts for `single_task()` return `PI_INVALID_WORK_DIMENSION` here until it is supported?